hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AExceptionTranslation.java

```diff
@@ -181,8 +182,12 @@ private static AmazonS3Exception createS3Exception(String message, int code,
 
   private static <E extends Throwable> E verifyTranslated(Class<E> clazz,
       AmazonClientException exception) throws Exception {
-    return verifyExceptionClass(clazz,
-        translateException("test", "/", exception));
+    // Verifying that the translated exception have the correct error message.
+    IOException ioe = translateException("test", "/", exception);
+    Assertions.assertThat(ioe.getMessage()).describedAs("Translated "
```

Comment:
if there is a mismatch, can you rethrow the entire exception. See (or use GenericTestUtils.assertExceptionContains()) 

```java
throw new AssertionError(<that message>, ioe);